### PR TITLE
boot: support attestation keybox provisioning

### PR DESCRIPTION
**Pull request by GangSecurity**
Tracked-On: OAM-91192
Signed-off-by: Chen Gang G <gang.g.chen@intel.com>

### CODE REVIEW

File: libfastboot/keybox_provision.c [hunk 1]
```diff
@@ -16,202 +16,77 @@
...
+#include <efi.h>
+#include <efilib.h>
+#include <gpt.h>
+#include <log.h>
 
-static struct gcm_key attkb_key = {0};
-static encrypted_attkb_t enc_kb;
-static UINT8 *start_kb_addr = NULL;
+#define MAX_KEYBOX_SIZE    16384
+#define KB_HEAD_OFFSET     512
```
Comment (yang8621) [inline libfastboot/keybox_provision.c:25]: after code clean, set it to 0

File: libfastboot/keybox_provision.c [hunk 2]
```diff
@@ -16,202 +16,77 @@
...
-	UINTN write_sz;
-	UINTN kb_meta_sz;
-	UINT8 *cipher_blob;
-	UINTN cipher_blob_sz;
-	UINT8 *addr;
-
-	if (size > MAX_KEYBOX_SIZE) {
-		error(L"keybox size exceeded limit");
-		return EFI_INVALID_PARAMETER;
+		efi_perror(ret, L"Could not write to Disk.\n");
```
Comment (James-Zzz) [inline libfastboot/keybox_provision.c:71]: Suggest to use: "Could not write keybox header to Disk" to locate error message easily

File: libfastboot/keybox_provision.c [hunk 3]
```diff
@@ -16,202 +16,77 @@
...
+				gpart.dio->WriteDisk,
+				5,
+				gpart.dio,
+				gpart.bio->Media->MediaId,
+				sizeof(keybox_header_t) + partoffset + KB_HEAD_OFFSET,
+				size,
+				data);
 	if (EFI_ERROR(ret)) {
-		error(L"keybox encryption failure!");
+		efi_perror(ret, L"Could not write to Disk.\n");
```
Comment (James-Zzz) [inline libfastboot/keybox_provision.c:84]: Suggest to use: "Could not write keybox data to Disk" to locate error message easily

Comment (jeremy-compostella) [reply libfastboot/keybox_provision.c:84]: Never add `\n` with `debug()`, `error()` or `efi_perror()` as these macro already append the `\n`.

**Pull request merged**